### ISSUE TITLE
Enum with ToString member crashes in string concatenation

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#38858]
@vertygo reported a crash in [another thread](https://github.com/dotnet/roslyn/issues/38725#issuecomment-534981484).
From investigating the crash dump file that he shared with me, I narrowed down a repro. The issue happens if you have an enum with a `ToString` member and use it in a string concatenation.

```C#
using System;
public class C 
{
    public void M(D c) 
    {
        _ = c + "";
    }
}
public enum D { A = 1, B = 2, ToString = 4 }
```

[sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJ9KBhS0gb1MteqpwBZKBZACgAilAMYBKBiTaVmkqWwD6lALwjKAakoAiTQG4WbAL6kjZKnAB2AVzCUhjSgEFllAkkoAhZ3jcAVAPYAyjAIaOYA5s7cBkA=)

**Comment by Leemyy** [comment]
This does not just happen with Enums.
The same crash occurs, if You have a `ToString` member on a struct:
```c#
public struct Bad
{
    new int ToString;

    string Crash()
    {
        return "" + this;
    }
}
```

**Comment by jcouv** [comment]
Fixed for 16.4 preview 2. Thanks for reporting!
